materialize-elasticsearch/schemabuilder/src/elastic_search_data_types.rs

```diff
@@ -0,0 +1,135 @@
+use super::errors::*;
+use serde::{Deserialize, Serialize};
+use serde_json::{json, map, Value};
+use std::collections::HashMap;
+
+// The basic elastic search data types to represent data in Flow.
+#[derive(Clone, Debug, Serialize, Deserialize)]
```

Comment:
I've make the changes. It simplified the code by using its default `type` tag.
One  issue is that now the `type` tag is associated with the enum variant. And seems it is inappropriate to override the `type` to be `text` for the enum `keyword`, b/c of [this](https://github.com/serde-rs/serde/issues/1161).

So I have restructured the enum types so that the dual type is associated with enum `Text`, which solved the issue, and make the enums more clear - each enum defined here has a direct counter-part in elastic search.